sql,stats: fix flakes in TestCacheWait and TestQueryCache

rytaft opened the pull request:
**stats: fix flake in TestCacheWait**

This commit fixes a flake in `TestCacheWait`, which was introduced
by #51616. That PR changed the call to `InvalidateTableStats` in
`TestCacheWait` to `RefreshTableStats`, but that change should not have
been made. The purpose of the test is to test that the cache invalidation
and waiting mechanisms work correctly, and therefore it must call
`InvalidateTableStats`, not `RefreshTableStats`.

Fixes #51712

**sql: fix flake in TestQueryCache/group/statschange**

This commit fixes a flake in TestQueryCache/group/statschange,
which was introduced by #51616. That PR made updates to the stats
cache asynchronous, so we can no longer expect the query cache to
be invalidated immediately after a stats update. This commit fixes
the problem by introducing a retry mechanism into the test.

Fixes #51693

rytaft committed:
stats: fix flake in TestCacheWait

This commit fixes a flake in TestCacheWait, which was introduced
by #51616. That PR changed the call to InvalidateTableStats in
TestCacheWait to RefreshTableStats, but that change should not have
been made. The purpose of the test is to test that the cache invalidation
and waiting mechanisms work correctly, and therefore it must call
InvalidateTableStats, not RefreshTableStats.

Fixes #51712

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/51828)
<!-- Reviewable:end -->

rytaft committed:
sql: fix flake in TestQueryCache/group/statschange

This commit fixes a flake in TestQueryCache/group/statschange,
which was introduced by #51616. That PR made updates to the stats
cache asynchronous, so we can no longer expect the query cache to
be invalidated immediately after a stats update. This commit fixes
the problem by introducing a retry mechanism into the test.

Fixes #51693

Release note: None

RaduBerinde reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/51828#-:-MCw9Tld2_qZjoUrtO9J:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @RaduBerinde)



<!-- Sent from Reviewable.io -->

rytaft commented:
TFTR!

bors r+





<!-- Sent from Reviewable.io -->

jordanlewis commented:
Courtesy re-bors due to flake in version-upgrade:

bors r+
